### PR TITLE
return credentials field pointing to omd vault plugin

### PR DESCRIPTION
**Pull request by cdoron**
which takes secrets directly from OM

Signed-off-by: Doron Chen <cdoron@il.ibm.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -94,7 +98,7 @@ func PrepareCmd() *cobra.Command {
 
 			err = yaml.Unmarshal(customizationFileBytes, &customization)
 			if err != nil {
-				logger.Error().Err(err).Msg("failure to parse customization file")
+				logger.Error().Err(err).Msg(FailureToParseCusromizationFile)
```
Comment (roytman) [inline main.go:101]: L:91 ->logger.Error().Err(err).Msg(FailureToReadConfigurationFile)

Can we replace lines 91-103 and 39-49 by a helper function ?

File: main.go [hunk 2]
```diff
@@ -34,13 +38,13 @@ func RunCmd() *cobra.Command {
 
 			configFileBytes, err := os.ReadFile(configFile)
 			if err != nil {
-				logger.Error().Err(err).Msg("failure to read config file")
+				logger.Error().Err(err).Msg(FailureToReadCustomizationFile)
```
Comment (roytman) [inline main.go:41]: FailureToRead**Configuration**File ?

File: main.go [hunk 3]
```diff
@@ -18,8 +19,31 @@ import (
 	"fybrik.io/openmetadata-connector/pkg/utils"
 )
 
+const Customization = "customization"
 const DefaultConfigFile = "/etc/conf/conf.yaml"
 const DefaultCustomizationFile = "./customization.yaml"
+const FailureToParseCustomizationFile = "failure to parse customization file"
```
Comment (roytman) [inline main.go:25]: can you put  `DefaultConfigFile` and `DefaultCustomizationFile` as a separate const statement ?
```golang
const (
     DefaultConfigFile = "/etc/conf/conf.yaml"
     DefaultCustomizationFile = "./customization.yaml"
)

const (
   Customization = "customization"
   FailureToParseCustomizationFile = "failure to parse customization file"
   FailureToReadConfigFile = "failure to read configuration file"
)
```

File: main.go [hunk 4]
```diff
@@ -18,8 +19,31 @@ import (
...
+const FailureToReadConfigFile = "failure to read configuration file"
+const FailureToReadCustomizationFile = "failure to read customization file"
+const FileContainingTagsAndPropertiesNeeded = "File containing tags and custom properties needed for working with Fybrik"
+
+func parseCustomizationFile(customizationFile string, logger *zerolog.Logger) (map[string]interface{}, bool) {
```
Comment (roytman) [inline main.go:30]: why do you return bool, and not error ?

File: main.go [hunk 5]
```diff
@@ -106,8 +113,7 @@ func PrepareCmd() *cobra.Command {
 			occ.PrepareOpenMetadataForFybrik(endpoint, user, password, customization, &logger)
 		},
 	}
-	cmd.Flags().StringVar(&customizationFile, "customization", customizationFile,
-		"File containing tags and custom properties needed for working with Fybrik")
+	cmd.Flags().StringVar(&customizationFile, Customization, customizationFile, FileContainingTagsAndPropertiesNeeded)
```
Comment (roytman) [inline main.go:116]: why do you print customizationFile twice  ?

Comment (cdoron) [reply main.go:116]: `&customizationFile` is where to place the value, whereas `customizationFile` is the default value.

**Review by roytman — APPROVED** [review]

**Pull request merged**